### PR TITLE
Fix non authenticated access to report

### PR DESCRIPTION
**Pull request by jesusfcr**
- Upgrade vue to latest 2 version
- Add target dev to dockerfile for development
- Redirect to login in case of api 401

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,15 +4,23 @@ FROM node:9-slim as builder
...
+COPY . .
 
 RUN npm run-script build
 
+FROM builder as dev
+
+# Node:9-slim is unable to add packages so we are using a node package instead.
+RUN npm install -g envsub
+
+ENV TEAMS_CRUD=true
```
Comment (seilagamo) [inline Dockerfile:20]: Why are you setting this as true?

Comment (jesusfcr) [reply Dockerfile:21]: Good question, this is related to `envsub -s dollar-basic config.json dist/config.json` in package.json. 
Here I m setting a default for that env variable in order to the envsub command to generate a valid json.
When starting the dev container we don't run run.sh where this value is set up.

I added a comment.

Comment (jesusfcr) [reply Dockerfile:21]: Also, I moved the envsub from package.json to the docker cmd to not interfere with the local usage.

**Review by seilagamo — CHANGES_REQUESTED** [review]

File: src/report/report.vue [hunk 2]
```diff
@@ -185,6 +185,25 @@ export default class LiveReport extends Vue {
       }
       return;
     }
+
+    // Using err instance of ResponseError doesn't work.
+    if (typeof err.response !== 'undefined') {
+      var re = err.response;
+      switch (re.status) {
+        case 401:
+          var redirectTo: string = window.location.toString().split("#")[0];
```
[the remark this answers is not in the document]
Comment (jesusfcr) [reply src/report/report.vue:194]: done, thanks

File: src/report/report.vue [hunk 3]
```diff
@@ -185,6 +185,25 @@ export default class LiveReport extends Vue {
       }
       return;
     }
+
+    // Using err instance of ResponseError doesn't work.
+    if (typeof err.response !== 'undefined') {
+      var re = err.response;
```
[the remark this answers is not in the document]
Comment (jesusfcr) [reply src/report/report.vue:191]: done, thanks

**Review by seilagamo — APPROVED** [review]
LGTM

**Pull request merged**